app/views/ShareMessageView/index.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useLayoutEffect, useState } from 'react';
+import { ScrollView, View } from 'react-native';
+import { StackNavigationOptions } from '@react-navigation/stack';
+import { RouteProp, StackActions, useNavigation, useRoute } from '@react-navigation/native';
+
+import { getPermalinkMessage } from '../../lib/methods';
+import { TGetCustomEmoji } from '../../definitions';
+import KeyboardView from '../../containers/KeyboardView';
+import scrollPersistTaps from '../../lib/methods/helpers/scrollPersistTaps';
+import I18n from '../../i18n';
+import * as HeaderButton from '../../containers/HeaderButton';
+import StatusBar from '../../containers/StatusBar';
+import { useTheme } from '../../theme';
+import { getUserSelector } from '../../selectors/login';
+import SafeAreaView from '../../containers/SafeAreaView';
+import { events, logEvent } from '../../lib/methods/helpers/log';
+import styles from './styles';
+import SelectPersonOrChannel from './SelectPersonOrChannel';
+import { useAppSelector } from '../../lib/hooks';
+import Message from '../../containers/message';
+import { NewMessageStackParamList } from '../../stacks/types';
+import { postMessage } from '../../lib/services/restApi';
+
+const ShareMessageView = () => {
+	const [roomsId, setRoomsId] = useState<string[]>([]);
+	const [sending, setSending] = useState(false);
+	const navigation = useNavigation();
+	const { theme, colors } = useTheme();
+
+	const {
+		params: { message }
+	} = useRoute<RouteProp<NewMessageStackParamList, 'ShareMessageView'>>();
+
+	const { blockUnauthenticatedAccess, server, serverVersion, user, baseUrl, Message_TimeFormat, customEmojis } = useAppSelector(
+		state => ({
+			user: getUserSelector(state),
+			server: state.server.server,
+			blockUnauthenticatedAccess: !!state.settings.Accounts_AvatarBlockUnauthenticatedAccess ?? true,
```

Comment:
Are you sure you need `!!`?

Reply:
make TS happy and keep only the boolean, not the TSettingValue